### DISCUSSION TITLE
Portfolio transaction loading

### DISCUSSION THREAD

**Discussion by azriel46d** [OpenBB-finance/OpenBB#4736] [Q&A]
Hi , from what i have gathered so far , portfolios need to a file stored as an XLSX or a CSV. is it possible to build it dynamically and pass in the dataframe rather than having a file path?

**Comment by priyanshujiiii** [comment]
You can store in sql format

**Comment by northern-64bit** [comment]
Hello @azriel46d!
Yes, this is possible for the SDK, but with some workarounds and it ***requires*** that you have downloaded the source (https://docs.openbb.co/terminal/installation/source).

Instead of using the usual way to load the portfolio (`p = openbb.portfolio.load("openbb_terminal/miscellaneous/portfolio/holdings_example.xlsx")`), you can use the following code:
```python
# Make sure to import PortfolioEngine from openbb_terminal/portfolio/portfolio_engine.py.
from openbb_terminal.portfolio.portfolio_engine import PortfolioEngine # This depends on where you have the code


risk_free_rate = 0
full_shares = False
benchmark_symbol = "SPY"

portfolio_engine = PortfolioEngine(transactions)
portfolio_engine.generate_portfolio_data()
portfolio_engine.set_risk_free_rate(risk_free_rate)
portfolio_engine.set_benchmark(symbol=benchmark_symbol, full_shares=full_shares)

p = portfolio_engine
```
Make sure that the dataframe, `transactions`, still follows the conventions described in the docs (https://docs.openbb.co/sdk/usage/intros/portfolio#portfolio-files).

I hope that this helps. Feel free to ask if you want a more detailed explanation, clarifications or just have other questions.
